manifests/tftp.pp

```diff
@@ -29,4 +29,75 @@
       root => $foreman_proxy::tftp_root,
     }
   }
+
+  case $::osfamily {
+    'RedHat': {
+      case $::operatingsystem {
+        'Fedora': {
+          if versioncmp($::operatingsystemrelease, '19') >= 0 or $::operatingsystemrelease == 'Rawhide' {
+            $grub_type = 'redhat'
+          } else {
+            $grub_type = 'redhat_old'
+          }
+        }
+        /^(RedHat|CentOS|Scientific|OracleLinux)$/: {
+          if versioncmp($::operatingsystemmajrelease, '7') >= 0 {
+            $grub_type = 'redhat'
+          } else {
+            $grub_type = 'redhat_old'
+          }
+        }
+      }
+    }
+    'Debian','Ubuntu': {
+      $grub_type = 'debian'
+    }
+  }
+
+  case $::operatingsystem {
+    'Fedora': {
+      $grub_efi_path = 'fedora'
+    }
+    'CentOS': {
+      $grub_efi_path = 'centos'
+    }
+    /^(RedHat|Scientific|OracleLinux)$/: {
+      $grub_efi_path = 'redhat'
+    }
+  }
+
+  $efi_dir = '/usr/lib/grub/x86_64-efi'
+  $grub_modules = 'all_video boot btrfs cat chain configfile echo efifwsetup efinet ext2 fat font gfxmenu gfxterm gzio halt hfsplus iso9660 jpeg loadenv loopback lvm mdraid09 mdraid1x minicmd normal part_apple part_msdos part_gpt password_pbkdf2 png reboot search search_fs_uuid search_fs_file search_label serial sleep syslinuxcfg test tftp video xfs linux backtrace usb usbserial_common usbserial_pl2303 usbserial_ftdi usbserial_usbdebug linuxefi'
```

Comment:
These modules will likely be different for Debian and EL6. Question is if I should parametrize this, users _might_ want to change that (to generate smaller binary for example).